### PR TITLE
Refactor Apply to ObjectSet

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

File: controllers/feature/controller.go [hunk 1]
```diff
@@ -175,3 +172,18 @@ func (f *featureHandler) isEnabled(name string) bool {
 	_, ok := f.featureState[name]
 	return ok
 }
+
+func (f *featureHandler) getDepFeature(depName, namespace string) (*v1.Feature, error) {
+	start := time.Millisecond * 250
```
Comment (ibuildthecloud) [inline controllers/feature/controller.go]: why is this needed?

File: cli/cmd/feature/feature.go [hunk 2]
```diff
@@ -3,6 +3,10 @@ package feature
 import (
 	"fmt"
 
+	"github.com/rancher/rio/cli/cmd/up"
+
```
Comment (ibuildthecloud) [inline cli/cmd/feature/feature.go]: whitespace

File: controllers/controllers.go [hunk 3]
```diff
@@ -3,6 +3,8 @@ package controllers
 import (
 	"context"
 
+	"github.com/rancher/rio/controllers/setting"
+
```
Comment (ibuildthecloud) [inline controllers/controllers.go]: whitespace

File: controllers/setting/setting.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
+package setting
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/rancher/rio/pkg/settings"
+
```
Comment (ibuildthecloud) [inline controllers/setting/setting.go]: whitespace

File: features/routing/controllers/routeset/controller.go [hunk 5]
```diff
@@ -3,6 +3,8 @@ package routeset
 import (
 	"context"
 
+	"github.com/rancher/rio/features/routing/controllers/util"
+
```
Comment (ibuildthecloud) [inline features/routing/controllers/routeset/controller.go]: whitespace

File: features/routing/controllers/service/controller.go [hunk 6]
```diff
@@ -3,6 +3,8 @@ package service
 import (
 	"context"
 
+	"github.com/rancher/rio/features/routing/controllers/util"
+
```
Comment (ibuildthecloud) [inline features/routing/controllers/service/controller.go]: whitespace

File: pkg/features/controller.go [hunk 7]
```diff
@@ -120,3 +121,7 @@ func (f *FeatureController) Start(ctx context.Context, feature *v1.Feature) erro
 
 	return nil
 }
+
+func (f *FeatureController) Priority() int {
+	return f.PriorityWeight
```
Comment (ibuildthecloud) [inline pkg/features/controller.go]: Why do we need a priority?

File: types/apis/project.rio.cattle.io/v1/zz_generated_deepcopy.go [hunk 8]
```diff
@@ -258,3 +258,61 @@ func (in *PublicDomainSpec) DeepCopy() *PublicDomainSpec {
 	in.DeepCopyInto(out)
 	return out
 }
+
```
Comment (ibuildthecloud) [inline types/apis/project.rio.cattle.io/v1/zz_generated_deepcopy.go]: can you seperate generated code to a different commit

File: controllers/feature/controller.go (hunk 1 above)
Comment (StrongMonkey) [reply controllers/feature/controller.go]: This is just to avoid `not found` error. The first try will always return `not found` error because the race condition when feature is not actually created.

File: pkg/features/controller.go (hunk 7 above)
Comment (StrongMonkey) [reply pkg/features/controller.go]: want to have better control over the order of features
